learing_journal/views/default.py

```diff
@@ -1,76 +1,121 @@
     },
 ]
 
 
-@view_config(route_name='home', renderer='templates/home.jinja2')
-def home(request):
+@view_config(route_name='private', renderer='string', permission='secret')
+def private(request):
+    return "I am a private view"
+
+
+@view_config(route_name='login', renderer='templates/login.jinja2', permission=NO_PERMISSION_REQUIRED)
+def login(request):
     if request.method == 'POST':
-        if request.POST['title'] != '' or request.POST['body'] != '':
-            title = request.POST['title']
-            body = request.POST['body']
-            month = time.strftime('%B')
-            day = time.strftime('%d')
-            year = time.strftime('%Y')
-            date = '{} {}, {}'.format(month, day, year)
-            new = MyModel(title=title, body=body, date=date)
-            request.dbsession.add(new)
-        else:
-            error_msg = "Can't submit empry entry"
-            return {'error_msg': error_msg}
+        username = request.params.get('username', '')
+        password = request.params.get('password', '')
+        if check_credentials(username, password):
+            headers = remember(request, username)
+            return HTTPFound(location=request.route_url('home'), headers=headers)
+    return {}
+
+
+@view_config(route_name='logout')
+def logout(request):
+    headers = forget(request)
+    return HTTPFound(request.route_url('home'), headers=headers)
+
+
+# TODO: test the routes
+# TODO: add if request.method == 'DELETE':
+@view_config(route_name='home', renderer='templates/home.jinja2', permission='view')
+def home(request):
     try:
-        query = request.dbsession.query(MyModel)
-        all_entries = query.all()
+        query = request.dbsession.query(Journal)
+        all_entries = query.order_by(desc(Journal.date)).all()
     except DBAPIError:
         return Response(db_err_msg, content_type='text/plain', status=500)
-    return {'entries': all_entries, 'poject': 'learning_journal'}
+    return {'entries': all_entries}
 
 
-@view_config(route_name='create', renderer='templates/new-entry.jinja2')
+@view_config(route_name='create',
+             renderer='templates/new-entry.jinja2',
+             permission='secret')
 def create(request):
-    return {'poject': 'learning_journal'}
+    '''handles the POST request to create a new entry'''
+    title = body = error = ''
 
+    if request.method == 'POST':
+        title = request.POST['title']
+        body = request.POST['body']
+        print('\nTitle: {}'.format(title))
+        print('Body: {}\n'.format(body))
 
-@view_config(route_name='update', renderer='templates/edit-entry.jinja2')
-@view_config(route_name='detail', renderer='templates/single-entry.jinja2')
+        if title != '' and body != '':
+            date = datetime.datetime.now()
+            date_last_updated = datetime.datetime.now()
```

Comment:
You could also simply bind both symbols to the same datetime object.  No need to do the job twice

```python
date = last_updated = datetime.datetime.now()
```